pkg/provisioner/filestorage/filestorage.go

```diff
@@ -0,0 +1,215 @@
+	glog.Infof("Looking up existing mount targets")
+	responseListMnt, err := fsp.client.FileStorage().ListMountTargets(ctx, filestorage.ListMountTargetsRequest{
+		AvailabilityDomain: ad,
+		CompartmentId:      &compartmentID,
+	})
+	if err != nil {
+		return nil, err
+	}
+	if len(responseListMnt.Items) != 0 {
+		glog.V(4).Infof("Found mount targets to use")
+		mntTargetSummary := responseListMnt.Items[rand.Int()%len(responseListMnt.Items)]
+		target, err := fsp.getMountTargetFromID(ctx, *mntTargetSummary.Id)
+		return target, err
+	}
+	// Mount target not created, create a new one
+	resp, err := fsp.client.FileStorage().CreateMountTarget(ctx, filestorage.CreateMountTargetRequest{
+		CreateMountTargetDetails: filestorage.CreateMountTargetDetails{
+			AvailabilityDomain: ad,
+			SubnetId:           &subnetID,
+			CompartmentId:      &compartmentID,
+			DisplayName:        common.String(fmt.Sprintf("%s%s", os.Getenv(volumePrefixEnvVarName), "mnt")),
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+	return &resp.MountTarget, err
+}
+
+func (fsp *filesystemProvisioner) Provision(options controller.VolumeOptions, ad *identity.AvailabilityDomain) (*v1.PersistentVolume, error) {
+	ctx, cancel := context.WithTimeout(fsp.client.Context(), fsp.client.Timeout())
+	defer cancel()
+
+	fileStorageClient := fsp.client.FileStorage()
+	response, err := fileStorageClient.CreateFileSystem(ctx, filestorage.CreateFileSystemRequest{
+		CreateFileSystemDetails: filestorage.CreateFileSystemDetails{
+			AvailabilityDomain: ad.Name,
+			CompartmentId:      common.String(fsp.client.CompartmentOCID()),
+			DisplayName:        common.String(fmt.Sprintf("%s%s", os.Getenv(volumePrefixEnvVarName), options.PVC.Name)),
+		},
+	})
+	if err != nil {
+		glog.Errorf("Failed to create a file system. Options:%#v, %s", options, err)
+		return nil, err
+	}
+
+	var target *filestorage.MountTarget
+	if options.Parameters[mntTargetID] == "" {
+		target, err = fsp.getOrCreateMountTarget(ctx, ad.Name, options.Parameters[subnetID], fsp.client.CompartmentOCID())
+	} else {
+		// Mount target already specified in the configuration file, find it in the list of mount targets
+		target, err = fsp.getMountTargetFromID(ctx, options.Parameters[mntTargetID])
+	}
+	if err != nil {
+		glog.Errorf("Failed to retrieve mount target: %s", err)
+		return nil, err
+	}
+
+	glog.V(4).Infof("Creating export set")
+	createExportResponse, err := fileStorageClient.CreateExport(ctx, filestorage.CreateExportRequest{
+		CreateExportDetails: filestorage.CreateExportDetails{
+			ExportSetId:  target.ExportSetId,
+			FileSystemId: response.FileSystem.Id,
+			Path:         common.String("/" + *response.FileSystem.Id),
+		},
+	})
+	if err != nil {
+		glog.Errorf("Failed to create export: %v", err)
+		return nil, err
+	}
+
+	var serverIP string
+	if len(target.PrivateIpIds) == 0 {
+		glog.Errorf("Failed to find server IDs associated with the mount target to provision a persistent volume")
```

Comment:
Needs some context adding to the log message (i.e. target ocid, pvc name perhaps). We'll migrate to the zap logger shortly but good to get in the habit of thinking about what contextual information you need in production.